cloudferrylib/os/actions/check_bandwidth.py

```diff
@@ -0,0 +1,97 @@
+# Copyright (c) 2015 Mirantis Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the License);
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an AS IS BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or
+# implied.
+# See the License for the specific language governing permissions and#
+# limitations under the License.
+
+
+import subprocess
+
+from fabric.api import env
+
+from cloudferrylib.base.action import action
+from cloudferrylib.utils import cmd_cfg
+from cloudferrylib.utils import utils
+
+
+LOG = utils.get_log(__name__)
+
+
+class CheckBandwidth(action.Action):
+    def run(self, **kwargs):
+        claimed_bandw = self.cloud.cloud_config.initial_check.claimed_bandwidth
+        test_file_size = self.cloud.cloud_config.initial_check.test_file_size
+        factor = self.cloud.cloud_config.initial_check.factor
+        req_bandwidth = claimed_bandw * factor
+        temp_file_name = 'bandwith_check.tmp'
+        local_file_path = '/tmp/' + temp_file_name
+        remote_file_path = '%s/%s' % (self.cloud.cloud_config.cloud.temp,
+                                      temp_file_name)
+
+        scp_upload = cmd_cfg.scp_cmd('',
+                                     'root',
+                                     self.cloud.cloud_config.cloud.ssh_host,
+                                     remote_file_path,
+                                     '/tmp/')
+
+        scp_download = cmd_cfg.scp_cmd(local_file_path,
+                                       'root',
+                                       self.cloud.cloud_config.cloud.ssh_host,
+                                       self.cloud.cloud_config.cloud.temp,
+                                       '')
+
+        # I guess it can be moved from here in other place
+        check_dir_cmd = "if [ ! -d {0} ]; then mkdir {0}; fi".format(
```

Comment:
`mkdir -p <path>` does this for you (see `mkdir --help`), so I guess you can move this command to `cmd_cfg.py`